app_package/core_func.py

```diff
@@ -28,6 +28,18 @@ def core_delete(class_name):
         if (class_to_delete is None):
             return 1
 
+        relations = Relationship.query.filter(class_name == Relationship.from_name).all()
+        for rel in relations:
+            db.session.delete(rel)
+
```

Comment:
These couple lines here are almost a duplicate were you intending to have the same thing twice

Reply:
These lines are meant to be nearly duplicated. The first deletes all entries in the relationship table where the class to be deleted is the parent, and the second deletes those which are children.

I'm on the lookout for a more efficient way to accomplish this; less ugly approaches have not been successful thus far.